app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,33 @@
+class SessionsController < Devise::SessionsController
+  include Devise::Controllers::Rememberable
+  def create
+    delete_legacy_cookie
+    super
+  end
+
+  def destroy
+    delete_legacy_cookie
+    cookies.delete :remember_user_token, domain: ".#{SiteConfig.app_domain}"
+    # Let's say goodbye to all the cookies when someone signs out.
+    cookies.clear(domain: cookie_domain)
+
+    super
+  end
+
+  private
+
+  def legacy_cookie_domain
+    Rails.env.production? ? SiteConfig.app_domain : nil
```

Comment:
This seems cleaner, especially if we add a guard clause to cookie domain (as asked below):

```suggestion
    Rails.env.production? && SiteConfig.app_domain
```

If there won't be a guard clause we can't make this change.